src/router.ts

```diff
@@ -10,8 +10,25 @@ import Proposal from '@/views/Proposal.vue';
 import User from '@/views/User.vue';
 import Create from '@/views/Create.vue';
 
+import { isAddress } from '@ethersproject/address';
+import { getNetwork } from '@/networks';
+
+const validateIdGuard = (to, from, next) => {
+  try {
+    const id = to.params.id;
+    const [networkId, spaceId] = id.split(':');
+    console.log(':beforeEnter,', networkId, spaceId);
+    getNetwork(networkId);
+    if (!isAddress(spaceId)) return next('/');
```

Comment:
This won't work for starknet spaces.

Reply:
So we don't need this validation then

Reply:
Maybe we do, we could validate both networks if we wanted. I don't know what we want, it was your idea I think :D

Reply:
I think we can add it later for both networks